### PR TITLE
 prov/verbs: Move SRQ code to EP/MSG code

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
This PR does the following:
1) The SRQ code was moved from verbs_srq.c file to verbs_msg_ep.c file.
This is needed for futher separation of common code into common
functions and re-use this common stuff.
2) Move common code of SRQ and MSG/EP to separate functions and re-use them
3) ibv_destory_srq returns negative value in case of failure.
The current implementation of SRQ fi_close ignores this value.
This patch uses return value of ibv_destroy_srq and returns
this velue as a result of fi_close

### CODE REVIEW

File: prov/verbs/src/verbs_msg_ep.c [hunk 1]
```diff
@@ -781,6 +772,198 @@ struct fi_ops_msg fi_ibv_msg_srq_ep_msg_ops = {
...
+
+	srq_ep = container_of(fid, struct fi_ibv_srq_ep, ep_fid.fid);
+	ret = ibv_destroy_srq(srq_ep->srq);
+	if (ret)
+		VERBS_WARN(FI_LOG_EP_CTRL,
+			   "Cannot destroy SRQ rc=%d\n", ret);
+
+	free(srq_ep);
+
+	return FI_SUCCESS;
```
[the remark this answers is not in the document]
Comment (gladkovdmitry17) [reply prov/verbs/src/verbs_msg_ep.c]: Sure, the `ret` has to be returned from the function.
Thanks!

[the remark this answers is not in the document]
Comment (gladkovdmitry17) [reply prov/verbs/src/verbs_msg_ep.c]: @sydidelot Thanks! I've applied this comment in the 3rd commit

**Comment by shefty** [comment]
Changes look fine.  But please work on describing *why* you are making each change as part of commit messages.

**Comment by gladkovdmitry17** [comment]
> But please work on describing why you are making each change as part of commit messages.

Ok, The comments were added into commit messages for each commit and description of the PR were updated as well.
So, waiting for the CI testing completion and merging this PR.
Thanks for the taking a look into this!

**Pull request merged**